test/framework/multi_template.go

```diff
@@ -0,0 +1,108 @@
+package framework
+
+import (
+	"context"
+	"fmt"
+	"regexp"
+	"testing"
+
+	vspherev1 "sigs.k8s.io/cluster-api-provider-vsphere/api/v1beta1"
+
+	"github.com/aws/eks-anywhere/pkg/api/v1alpha1"
+	"github.com/aws/eks-anywhere/pkg/logger"
+	"github.com/aws/eks-anywhere/pkg/providers/common"
+)
+
+const (
+	// VSphereMultiTemplateUbuntu127 is to test multiple vSphere templates.
+	VSphereMultiTemplateUbuntu127 = "T_VSPHERE_MULTI_TEMPLATE_UBUNTU_1_27"
+	// ControlPlaneMachineLabel is to get control plane vSphere machines from machine list.
+	ControlPlaneMachineLabel = "cluster.x-k8s.io/control-plane"
+	// EtcdMachineLabel is to get etcd vSphere machines from machine list.
+	EtcdMachineLabel = "cluster.x-k8s.io/etcd-cluster"
+	// workerMachineLabel is to get worker vSphere machines from machine list.
+	workerMachineLabel = "cluster.x-k8s.io/deployment-name"
+)
+
+// VSphereMultiTemplateUbuntu127RequiredEnvVars is required for TestVSphereMultipleTemplatesUbuntu127.
+var VSphereMultiTemplateUbuntu127RequiredEnvVars = []string{
+	VSphereMultiTemplateUbuntu127,
+}
+
+// RequiredVsphereMultiTemplateUbuntu127EnvVars return required env vars for TestVSphereMultipleTemplatesUbuntu127.
+func RequiredVsphereMultiTemplateUbuntu127EnvVars() []string {
+	return VSphereMultiTemplateUbuntu127RequiredEnvVars
+}
+
+// CheckVsphereMultiTemplateUbuntu127EnvVars checks is required env vars are present.
+func CheckVsphereMultiTemplateUbuntu127EnvVars(t *testing.T) {
+	checkRequiredEnvVars(t, VSphereMultiTemplateUbuntu127RequiredEnvVars)
```

Comment:
If we are anyways checking it here, we don't need the setup logic right? If a user doesn't specify, it should only fail for this test and not for other tests I think?

Reply:
yes, so it's only called in the test `TestVSphereMultipleTemplatesUbuntu127`